### PR TITLE
FPCore to Mathematica (Wolfram Language) compiler

### PR DESCRIPTION
**Pull request by billzorn**
This compiler ignores all precision annotations and converts FPCores into Wolfram Language as directly as possible. Infinity and NaN from floating-point land are translated into Infinity and Indeterminate, which don't necessarily have the same behavior. This is not meant to be a rigorous translation of the subtleties of floating-point edge cases and rounding behavior, but rather a low-overhead translation of simple mathematical expressions.

This inital version of the compiler is not complete or particularly well tested. Loops will be added soon.

It is not immediately obvious how to do automated testing, as proper tests require Mathematica to be installed (and thus a Mathematica license...). There are also some potentially tricky edge cases, for example with Mathematica continuing computations with imaginary numbers.

### CODE REVIEW

File: tools/core2math.rkt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    ;['isnormal ""]
+    ;['signbit "(Sign[~a] == -1)"]
+    [_ (error 'operator->math "Unsupported operator ~a" op)]))
+
+(define (application->math operator args)
+  (match (cons operator args)
+    [(list (or '< '> '<= '>= '== '!= 'and 'or) args ...)
+     (format (operator->math operator)
+             (string-join
+              (for/list ([a args]) (format "~a" a))
```
Comment (pavpanchekha) [inline tools/core2math.rkt]: This is just `(map ~a a)`

**Comment by pavpanchekha** [comment]
The code looks fine. I would push to name it something like `core2wls` or similar; `math` might mean something other than Mathematica to some people :).

As for automated testing, I suppose we will simply have to go without automated testing. I'd appreciate it if you could write an automated tester, and gate it behind `which math` or something, though, so we can at least run the tests locally.

The challenge with imaginary numbers is also tricky, but can perhaps be resolved by allowing non-NaN results when the original code returns NaN. In any case, it is likely what the user wants.

**Comment by billzorn** [comment]
Thanks Pavel for the input.

The "tests" are not particularly rigorous - they ignore cases where the reference interpreter produces a NaN, or where Mathematica produces output that does not look like a number. There is also no attempt to coordinate precision: the reference interpreter will use the provided binary32 or binary64 annotation, while Mathematica ignores it and just computes the result numericallt as f(args) // N. The tests are expected to expose a lot of discrepancies in the behavior, but they give some confidence that the generated wolframscript expressions are at least syntactically correct.

I expect to merge this tomorrow, after taking another look at the currently unsupported operations.

**Comment by pavpanchekha** [comment]
Looks good to me!
